test/UnitTest.cs

```diff
@@ -1934,6 +1934,35 @@ public void TestExpr()
             TU.ExecuteSQL(sql, "0,1,2,3");
             sql = "select a1 from a where not not not a1 in (1)";
             TU.ExecuteSQL(sql, "0;2");
+
+            // From issue #35, some of the failing ones are passing now
+            // BUG: Incorrect results but no longer insists that a1 show up in group by list
+            sql = "select abs(-a1*2), count() from a group by round(a1, 10);";
+            // There are two more bugs: count() should raise error, results should be the following
```

Comment:
which # tracking these issues?

Reply:
abs(-a1 * 2) is in issue #35. I have added it to unit tests.
I have also added:
select (select a1 from a order by -a1 limit 1), count(a1) from a group by (select a1 from a order by -a1

I tried to add
select a1, (select count(b2) from b where b1=a1 and b2>2), (select count(b2) from b where b1=a1 and b3>3) from a;

but it was returning different output than what I was asserting, no matter what. So I disabled it.

I forgot to add the following negative tests. They are returning error, which is the correct behavior.

select a1, sum(a1) from a group by a1 having sum(a2) > a3;
select * from a having sum(a2) > 1;

Reply:
I have opened #223 to triage executor related issues. If you think the issue is out of the algebrizer/optimizer's range, let's put them in #233.

Reply:
I think
select a1, (select count(b2) from b where b1=a1 and b2>2), (select count(b2) from b where b1=a1 and b3>3) from a;
and
select (select a1 from a order by -a1 limit 1), count(a1) from a group by (select a1 from a order by -a1
may or may not be executor issues.

What is noticed in "select abs(-a1*2), count(*) from a group by round(a1, 10)" is that the value fed into abs() is not (-a1 * 2), instead it is round(a1, 10). I think the same thing happens in the two statements above. So it may be group by ordinal is wrong. I will look into it again. For now, may be they can be in issue #35.